app/os_android.go

```diff
@@ -951,3 +959,59 @@ func Java_org_gioui_Gio_scheduleMainFuncs(env *C.JNIEnv, cls C.jclass) {
 }
 
 func (_ ViewEvent) ImplementsEvent() {}
+
+var foregroundService struct {
+	once   sync.Once
+	intent C.jobject
+	mu     sync.Mutex
+	count  int
+}
+
+// StartForeground starts the foreground service
+func StartForeground(title, text string) (err error) {
+	foregroundService.mu.Lock()
+	defer foregroundService.mu.Unlock()
+	foregroundService.once.Do(func() {
```

Comment:
You don't need a sync.Once when using a mutex. Just check that foregroundService.count is 0.